geode/src/main/java/org/apache/calcite/adapter/geode/rel/GeodeSchemaFactory.java

```diff
@@ -39,18 +46,58 @@
   public static final String PDX_SERIALIZABLE_PACKAGE_PATH = "pdxSerializablePackagePath";
   public static final String ALLOW_SPATIAL_FUNCTIONS = "spatialFunction";
   public static final String COMMA_DELIMITER = ",";
+  public static final String JNDI_INITIAL_CONTEXT_FACTORY = "jndiInitialContextFactory";
+  public static final String JNDI_CLIENT_CACHE_OBJECT_KEY = "jndiClientCacheObjectKey";
 
   public GeodeSchemaFactory() {
     // Do Nothing
   }
 
+  private static Context getContext(Map map) {
+    Context context = null;
+
+    try {
+      String jndiInitialContextFactory = (String) map.get(JNDI_INITIAL_CONTEXT_FACTORY);
```

Comment:
What if `JNDI_INITIAL_CONTEXT_FACTORY` is missing ?

Reply:
Will check for null and return null